test/indent_anchor_test.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require_relative 'lib'
```

Comment:
The test cases are nice, but I would try to be more methodical when testing APIs that contain values:

1. I would test a simple group, with a single break.
  - Try indentation values of 0, 1, and 2.
  - Assert that it would break with a negsative indentation.
1. I would start layering my examples and elaborating them, making them at the same time more complex, but also easier to grok by future me or any other maintainer who'd take a look at the tests:
   1. Add an empty group, inside the outer group, vary the indentation, make sure nothing breaks
   1. Now add some text only in the group, make sure I get what I expect, and also vary the indentations
   1. Now continue elaborating, and add a break inside it, before the text …
   1. Add a break after the text.
   1. Add a break before the nested group after some text
   1. Add a break before in between text, before the inner group …
   1. Add a text after the group, in all different configs we just saw
   1. Add a break …
   1. Add breaks before and after …
   1. etc … you get my point.
   
And don't forget that these tests will be needed, to test `nest` when it lands.

So it's worthwhile to sit down and think about the best way to do it.